File: spark-study-sql-java/src/main/java/com/wen/spark/sql/core/ParquetDiscovery.java
package com.wen.spark.sql.core;

import org.apache.hadoop.conf.Configuration;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.DataFrameReader;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SQLContext;

import java.util.List;

public class ParquetDiscovery {
    public static void main(String[] args) {
        SparkConf conf=new SparkConf().setAppName("ParquetDiscovery").setMaster("local");
        JavaSparkContext sc=new JavaSparkContext(conf);
        System.setProperty("HADOOP_USER_NAME", "root");
        sc.hadoopConfiguration().set("dfs.client.use.datanode.hostname","true");
        SQLContext sqlContext=new SQLContext(sc);
        DataFrameReader reader=sqlContext.read();
      //  Dataset ds= reader.json("hdfs://hadoop:8020/data/gender=male/country=us/users.parquet");
         Dataset ds= reader.format("parquet").load("hdfs://cloud.codeguoj.cn:8020/test/parquet/");
        //
        ds.printSchema();
        ds.show();
        sc.close();
    }
}

File: spark-study-sql-scala/src/main/java/com/wen/spark/sql/core/ParquetDiscovery.scala
package com.wen.spark.sql.core

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.api.java.JavaSparkContext
import org.apache.spark.sql.{DataFrameReader, SQLContext}

object ParquetDiscovery {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("ParquetDiscovery")
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)
    val reader = sqlContext.read
    val ds = reader.json("hdfs://hadoop:8020/data/gender=male/country=us/users.parquet")
    ds.printSchema()
    ds.show()
  }
}
